File: src/main/scala/lesson/scala/basics/Loop.scala
package lesson.scala.basics

object Loop {
  def main(args: Array[String]) = {
    val intArray = (1 to 10).toArray

    //for
    for(i <- 0 until intArray.size) println(intArray(i))

    //foreach
    for(i <- intArray) println(i)
    intArray.foreach(x => println(x))
    intArray.foreach(println(_))
  }
}

File: src/main/java/lesson/java/basics/Loop.java
package lesson.java.basics;

/**
 * Created with IntelliJ IDEA.
 * User: ippei
 * Date: 6/9/13
 * Time: 3:13 PM
 * To change this template use File | Settings | File Templates.
 */
public class Loop {
    public static void main(String args[]){
        final int[] intArray = {1,2,3,4,5,6,7,8,9,10};

        // for
        for (int i = 0; i < intArray.length; i++){
            System.out.println(String.valueOf(intArray[i]));
        }

        // foreach
        for(int i : intArray){
            System.out.println(String.valueOf(i));
        }
    }
}
